### PR TITLE
feat: Added setup and pipeline steps including docker and tests to solution

### PR DESCRIPTION
**Pull request by JesperGomes**
Added setup and pipeline steps including docker and tests to solution.
The test/fixture section is not complete.

- [x] The title adheres to [this guide](https://github.com/Mech0z/GitHubGuidelines)
- [x] Tests are written and executed locally
- [ ] Subsystem tests have been tested (by manually deploying to `dev_002`)

### CODE REVIEW

File: .github/workflows/cd.yml [hunk 1]
```diff
@@ -47,7 +47,7 @@ jobs:
 
   electrical_heating:
     needs: changes
-    if: ${{ needs.changes.outputs.electrical_heating == 'true' }}
+    if: ${{ needs.changes.outputs.any == 'true' }}
```
Comment (johevemi) [inline .github/workflows/cd.yml:50]: should this be electrical_heating_job instead of any?

File: .github/workflows/ci-orchestrator.yml [hunk 2]
```diff
@@ -26,21 +26,22 @@ jobs:
...
-    uses: Energinet-DataHub/.github/.github/workflows/python-build-and-push-docker-image.yml@v13
+    uses: Energinet-DataHub/.github/.github/workflows/python-build-and-push-docker-image.yml@v14
     with:
       docker_changed: ${{ needs.changes.outputs.docker == 'true' }}
       docker_changed_in_commit: ${{ needs.changes.outputs.docker_in_commit == 'true' }}
 
   ci_electrical_heating:
     needs: [changes, ci_docker]
-    if: ${{ needs.changes.outputs.electrical_heating == 'true' }}
+    if: ${{ needs.changes.outputs.any == 'true' }}
```
Comment (johevemi) [inline .github/workflows/ci-orchestrator.yml:36]: same question: electrical_heating_job?

File: .github/workflows/cd.yml (hunk 1 above)
Comment (energinet-ajw) [reply .github/workflows/cd.yml:50]: Don't know. Ask Nicolaj.

File: .github/workflows/ci-orchestrator.yml (hunk 2 above)
Comment (energinet-ajw) [reply .github/workflows/ci-orchestrator.yml:36]: Stop, stop. Ask Nicolaj.

**Review by johevemi — APPROVED** [review]

File: .github/workflows/cd.yml (hunk 1 above)
Comment (johevemi) [reply .github/workflows/cd.yml:50]: its fine. I dont think it matters for now. We can change it when adding effect payment

File: .github/workflows/ci-orchestrator.yml (hunk 2 above)
Comment (NicolajAaH) [reply .github/workflows/ci-orchestrator.yml:36]: I believe it should be changed back. Not sure why it is edited

**Review by johevemi — DISMISSED** [review]

**Review by NicolajAaH — APPROVED** [review]

File: source/electrical_heating/contracts/parameters-reference.txt [hunk 3]
```diff
@@ -0,0 +1,8 @@
+# This file contains all the parameters that the job consumers. Some are required and some or not.
+#
+# Empty lines and lines starting with '#' are ignores in the tests.
+
+# Required parameters
```
Comment (JesperGomes) [inline source/electrical_heating/contracts/parameters-reference.txt:5]: Wring parameter name? Should be orchestrator_instance_id?

File: .devcontainer/check_test_count.sh
Comment (JesperGomes) [inline .devcontainer/check_test_count.sh:1]: Is this file needed? If so, should it be called something specific for electrical heating?

File: source/electrical_heating/contracts/parameters-reference.txt (hunk 3 above)
Comment (energinet-ajw) [reply source/electrical_heating/contracts/parameters-reference.txt:5]: Corrected.

File: .devcontainer/check_test_count.sh
Comment (energinet-ajw) [reply .devcontainer/check_test_count.sh:1]: It is used by ci-electrical-heating.yml

File: source/electrical_heating/contracts/parameters-reference.txt
Comment (BjarkeMeier) [inline source/electrical_heating/contracts/parameters-reference.txt:1]: How about renaming the file to `job-parameters.txt`?

File: source/electrical_heating/contracts/parameters-reference.txt [hunk 4]
```diff
@@ -0,0 +1,7 @@
...
+--orchestration-instance-id={orchestration-instance-id}
+--created-by-user-id={created-by-user-id}
```
Comment (BjarkeMeier) [inline source/electrical_heating/contracts/parameters-reference.txt:7]: Actor ID

Comment (energinet-ajw) [reply source/electrical_heating/contracts/parameters-reference.txt]: Changed.

File: source/electrical_heating/contracts/parameters-reference.txt
Comment (energinet-ajw) [reply source/electrical_heating/contracts/parameters-reference.txt:1]: I kept the former.

**Review by energinet-ajw — DISMISSED** [review]

File: .devcontainer/check_test_count.sh [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+# It is used in the pipeline to ensure that all the tests are included in the matrix.
+# The script must be invoked with a filter matching the paths NOT included in the matrix
+
+# $@: (Optional) Can be set to specify a filter for running python tests at the specified path.
+echo "Filter (paths): '$@'"
+
+# Exit immediately with failure status if any command fails
+set -e
+
+cd source/electrical_heating/tests/
```
Comment (ShaBloze) [inline .devcontainer/check_test_count.sh:14]: as we will probably have more packages with tests, should the `electrical_heating` be a parameter?

File: .github/workflows/detect-changes.yml [hunk 6]
```diff
@@ -37,9 +37,14 @@ jobs:
         id: filter
         with:
           filters: |
+            any:
```
Comment (ShaBloze) [inline .github/workflows/detect-changes.yml:40]: is `any` a good name here? it does not seem like it looks for 'any' changes

**Review by NicolajAaH — DISMISSED** [review]

File: .devcontainer/check_test_count.sh (hunk 5 above)
Comment (energinet-ajw) [reply .devcontainer/check_test_count.sh:14]: When we implement "effekt betaling" I/we will cover this issue.

File: .github/workflows/detect-changes.yml (hunk 6 above)
Comment (energinet-ajw) [reply .github/workflows/detect-changes.yml:40]: We will fix this.

**Review by johevemi — DISMISSED** [review]

**Review by NicolajAaH — APPROVED** [review]

File: source/electrical_heating/contracts/job-parameters.txt
Comment (BjarkeMeier) [inline source/electrical_heating/contracts/job-parameters.txt:1]: Dont' delete?

File: source/electrical_heating/contracts/parameters-reference.txt
Comment (BjarkeMeier) [inline source/electrical_heating/contracts/parameters-reference.txt:1]: Remove?

File: source/electrical_heating/contracts/job-parameters.txt
Comment (energinet-ajw) [reply source/electrical_heating/contracts/job-parameters.txt:1]: Yes,

File: source/electrical_heating/contracts/parameters-reference.txt
Comment (energinet-ajw) [reply source/electrical_heating/contracts/parameters-reference.txt:1]: No

**Review by NicolajAaH — APPROVED** [review]

**Review by johevemi — APPROVED** [review]
